src/Microsoft.AspNetCore.JsonPatch/Internal/DictionaryAdapterOfTU.cs

```diff
@@ -133,11 +138,50 @@ private object ConvertValue(IDictionary<string, object> dictionary, string key,
                     var conversionResult = ConversionResultProvider.ConvertTo(newValue, existingValue.GetType());
                     if (conversionResult.CanBeConverted)
                     {
-                        return conversionResult.ConvertedInstance;
+                        var convertedInstance = conversionResult.ConvertedInstance;
+                        return TryParse<TValue>(convertedInstance);
                     }
                 }
             }
             return newValue;
         }
+
+        private static T TryParse<T>(string value)
```

Comment:
@rynowak Can we talk about what's the best way to either combine the two `TryParse` methods or if there is a better way to convert keys and values to `TKey` and `TValue`?